src/main/java/com/rackspacecloud/metrics/queryservice/controllers/QueryController.java

```diff
@@ -36,8 +36,9 @@
      */
     @GetMapping("/grafana-query")
     @Timed(value = "query.service", extraTags = {"query.type","query.grafana"})
+    // TODO: test-generator does not provide numeric tenant ids so this validation will fail for test data
     public QueryResult query(
-            final @RequestParam(value = "db", required = true) @Valid @Pattern(regexp="-?\\d+") String dbName, //dbName = tenantId
+            final @RequestParam(value = "db", required = true) /*@Valid @Pattern(regexp="-?\\d+")*/ String dbName, //dbName = tenantId
```

Comment:
And tenant IDs might ending up needing to convey an "account type" prefix like "hybrid:"